From ce5c40cca069aac565bc9def2251eb68b64ab3f9 Mon Sep 17 00:00:00 2001
From: Ken Ma <make@marvell.com>
Date: Mon, 28 Aug 2017 15:08:51 +0800
Subject: [PATCH 1954/2241] ARM64: dts: marvell: Fix south bridge pins
 definition for for Armada 37xx

Previously pcie group used smi group selection bitmask(bit4) wrongly,
but smi group was not added in dts; and pcie pins are used as gpio
actually, so pcie group's function should be set as "gpio" but not
"pcie".
This patch adds smi pins group definition for Armada 37xx DB and
community boards and sets pcie group function to "gpio".

Change-Id: I7220197a63a9500d7b4d3742435b7bb43f1a293a
Signed-off-by: Ken Ma <make@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/43444
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-3720-community.dts | 2 ++
 arch/arm64/boot/dts/marvell/armada-3720-db.dts        | 2 +-
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi          | 7 ++++++-
 3 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-community.dts b/arch/arm64/boot/dts/marvell/armada-3720-community.dts
index cc68158..992c4c9 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-community.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-community.dts
@@ -77,6 +77,8 @@
 			};
 
 			eth0: ethernet@30000 {
+				pinctrl-names = "default";
+				pinctrl-0 = <&rgmii_pins>, <&smi_pins>;
 				phy-mode = "rgmii-id";
 				status = "okay";
 				buffer-manager = <&bm>;
diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index 318e7b2..7ce8868 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -133,7 +133,7 @@
 
 			eth0: ethernet@30000 {
 				pinctrl-names = "default";
-				pinctrl-0 = <&rgmii_pins>;
+				pinctrl-0 = <&rgmii_pins>, <&smi_pins>;
 				phy-mode = "rgmii-id";
 				phy = <&phy0>;
 				status = "okay";
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index e185344..e6f170a 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -330,6 +330,11 @@
 					function = "mii";
 				};
 
+				smi_pins: smi-pins {
+					groups = "smi";
+					function = "smi";
+				};
+
 				sdio_pins: sdio-pins {
 					groups = "sdio_sb";
 					function = "sdio";
@@ -337,7 +342,7 @@
 
 				pcie_pins: pcie-pins {
 					groups = "pcie1";
-					function = "pcie";
+					function = "gpio";
 				};
 			};
 
-- 
2.7.4

